src/native/windows/src/service.c

```diff
@@ -575,6 +575,57 @@ apxServiceControl(APXHANDLE hService, DWORD dwControl, UINT uMsg,
     return FALSE;
 }
 
+BOOL
+apxServiceCheckStop(APXHANDLE hService)
+{
+    LPAPXSERVICE   lpService;
+    SERVICE_STATUS stStatus;
+    DWORD          dwState = SERVICE_STOPPED;
+    DWORD          sleepMillis;
+
+    if (hService->dwType != APXHANDLE_TYPE_SERVICE)
+        return FALSE;
+
+    lpService = APXHANDLE_DATA(hService);
+    /* Manager mode cannot handle services */
+    if (lpService->bManagerMode) {
+        apxLogWrite(APXLOG_MARK_ERROR "apxServiceCheck(): Manager mode cannot handle services, returning FALSE");
+        return FALSE;
+    }
+    /* Check if the ServiceOpen has been called */
+    if (IS_INVALID_HANDLE(lpService->hService)) {
+        apxLogWrite(APXLOG_MARK_ERROR "apxServiceCheck(): Service is not open, returning FALSE");
+        return FALSE;
+    }
+
+    /* Check if we are in the stopped state */
+    sleepMillis = 1000;
+    apxLogWrite(APXLOG_MARK_DEBUG "apxServiceCheck(): Sleeping %d milliseconds", sleepMillis);
+    Sleep(sleepMillis);
+
+    if (QueryServiceStatus(lpService->hService, &stStatus)) {
+        apxLogWrite(APXLOG_MARK_DEBUG "apxServiceCheck(): QueryServiceStatus OK");
+        if (stStatus.dwCurrentState == dwState) {
+            return TRUE;
+        } else {
+            apxLogWrite(APXLOG_MARK_DEBUG
```

Comment:
We have an API that does this kind of logging in prunsrv, refactor and reuse?

Reply:
API that does that? Where?

Reply:
Never mind; on 2nd, thought it is not be worth refactoring the call to apxLogWrite() in reportServiceStatusE(), since the message format here is a little different.